lib/sinon/spy.js

```diff
@@ -53,9 +53,11 @@ function matchingFake(fakes, args, strict) {
         return undefined;
     }
 
-    for (var i = 0, l = fakes.length; i < l; i++) {
-        if (fakes[i].matches(args, strict)) {
-            return fakes[i];
+    var fakesIndex = fakes.length;
+
+    while (fakesIndex--) {
```

Comment:
You're right, `|| undefined` is not necessary.

It'd be great to get the changes in, thanks!